astropy/visualization/wcsaxes/axislabels.py

```diff
@@ -23,17 +24,23 @@ def __init__(self, frame, minpad=1, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.set_clip_on(True)
         self.set_visible_axes("all")
-        self.set_ha("center")
-        self.set_va("center")
-        self._minpad = minpad
-        self._visibility_rule = "labels"
+        self.set_minpad(minpad)
+        self.set_loc(loc)
+        self.set_rotation_mode("anchor")
+        self.set_visibility_rule("labels")
```

Comment:
please check this thoroughly, but I'd expect this could be wrapped as a single method call as follows

```suggestion
        self.set(
            minpad=minpad,
            loc=loc,
            rotation_mode="anchor",
            visibility_rule="labels",
        )
```